Sources/MongoSwiftSync/MongoClient.swift

```diff
@@ -160,7 +160,7 @@ public class MongoClient {
      * - Returns: a `MongoDatabase` corresponding to the provided database name
      */
     public func db(_ name: String, options: DatabaseOptions? = nil) -> MongoDatabase {
-        return MongoDatabase(name: name, client: self, options: options)
+        return MongoDatabase(asyncDB: self.asyncClient.db(name, options: options))
```

Comment:
the change to this initializer is for consistency with how I ended up doing it for `MongoCollection`. I did it differently there because we can end up with a sync `MongoCollection` via both `createCollection` and `collection` and so I wanted to just be able to pass an async collection into the init.

I also realized there is no need for the sync DBs and collections to store their parent clients so that eliminated a need to pass that in